test/metabase/test/util/timezone.clj

```diff
@@ -19,19 +18,17 @@
       (thunk)
       (do
         (mb.hawk.parallel/assert-test-is-not-parallel "with-system-timezone-id")
-        ;; only if the app DB is already set up, we need to make sure plugins are loaded and kill any connection pools that
-        ;; might exist
-        (when (initialize/initialized? :db)
-          (initialize/initialize-if-needed! :plugins)
-          (#'driver/notify-all-databases-updated))
-        (try
-          (TimeZone/setDefault new-timezone)
-          (System/setProperty "user.timezone" timezone-id)
-          (t/testing (format "JVM timezone set to %s" timezone-id)
-            (thunk))
-          (finally
-            (TimeZone/setDefault original-timezone)
-            (System/setProperty "user.timezone" original-system-property)))))))
+        ;; Pre-emptively __wrapping__ try block in [[driver.tu/wrap-notify-all-databases-updated]]. I'm not sure that
+        ;; is currently necessary and I'm doing that to avoid surprises as with issue #36852 in future.
```

Comment:
Now connections are cleaned up also after `do-with-system-timezone-id`. Comment tries to explain why.